apps/aecore/test/aec_chain_sim.erl

```diff
@@ -407,28 +429,31 @@ fork_switch_(ForkId, #{forks := Forks, mempool := Pool, orphans := Orphans} = Ch
     NewChain = Chain#{ forks => NewForks
                      , mempool => NewPool
                      , orphans => Evict ++ Orphans },
-    {ok, announce(main, [], NewChain)}.
+    {ok, announce(main, [], NewChain, Opts)}.
 
 %% Announce top_changed and tx events
-announce(ForkId, Txs, #{ forks := Forks } = Chain) ->
+announce(ForkId, Txs, #{ forks := Forks } = Chain, Opts) ->
     #{ ForkId := #{ blocks := [#{ hash := TopHash
                                 , prev := PrevHash
                                 , header := Hdr } | _] = Blocks} } = Forks,
+    Simulator = simulator(Opts),
     Height = length(Blocks) + 1,
     Type = aec_headers:type(Hdr),
     Origin = origin(Type),
-    Info = #{ block_hash   => TopHash
+    Info = #{ pid => self(),
+              block_hash   => TopHash
             , block_type   => Type
             , block_origin => Origin
             , prev_hash    => PrevHash
             , height       => Height },
-    send_tx_events(Txs, Info),
-    case ForkId of
-        main ->
-            ?LOG("Publishing top_changed, I = ~p", [Info]),
-            aec_events:publish(top_changed, Info);
-        _ ->
-            ignore
+    if Simulator == parent_chain ->
+        aec_events:publish({parent_chain, top_changed}, Info#{txs => Txs});
```

Comment:
Commitments for the new block will already be stored in the DB - this event should inform us about the current top hash so we may check if we are following the correct fork on parent and in some edge cases re fetch missing blocks.

Reply:
Yep, as we discussed at the last time. But we have to support the common block type (for query and for arrived event too). Otherwise proto can be smashed (for now all the proto are **tx** and **block** abstract types). We can left it as empty list by my suggestion is to accept the all available data from client's event.